### ISSUE TITLE
Allow form data to be attached to each task when uploading

### ISSUE THREAD

**Issue by luluseventhbeam** [781flyingdutchman/background_downloader#28]
I have a use case, where I need to be able to add form data to the multipart/form-data upload request. 

We have a lot of images that will be uploaded and each has associated data that needs to be sent with it.

It would be great if the UploadTask object could have a Map<String, String> property for adding this.

**Comment by 781flyingdutchman** [comment]
Hi @luluseventhbeam, I've implemented this on the `upload_fields` branch and before I publish it, I was wondering if you could test the functionality and confirm it works as you expect it to?  To use the unpublished version with this feature, change your pubspec.yaml entry to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: upload_fields
```
To add fields to your `UploadTask`, simply set `fields` to a `Map<Sgtring, String>` with name/value pairs. This only works for multi-part uploads, not for binary uploads (where you set `post` to 'binary') but that doesn't seem to be your use case.

**Comment by luluseventhbeam** [comment]
Hi @781flyingdutchman ,

Is it possible to also pass through the field name for the data as a parameter?
Our backend dev mentioned the field name for the data is coming through as “file”, we need it to be “PhotoFile, I think once thats done it should all work 🙂

**Comment by 781flyingdutchman** [comment]
Added `fileField` (and `mimeType`) optional parameters to the `UploadTask` that allow you to do that. Please let me know if this addresses the issue.

**Comment by 781flyingdutchman** [comment]
Thanks for your help developing this! Published as 5.4.4.

**Comment by hadi7khan** [comment]
Can we download an image and pass data to it? Like i have a post request for downloading an image but can i also pass body with it? {imageName: "image.jpg"} something like this?
